src/unitxt/task.py

```diff
@@ -10,42 +13,141 @@ class Tasker:
 class FormTask(Tasker, StreamInstanceOperator):
     """FormTask packs the different instance fields into dictionaries by their roles in the task.
 
+    Attributes:
+        inputs (Union[Dict[str, Any], List[str]]):
+            Dictionary with string names of instance input fields and types of respective values.
+            In case a list is passed, each type will be assumed to be Any.
+        outputs (Union[Dict[str, Any], List[str]]):
+            Dictionary with string names of instance output fields and types of respective values.
+            In case a list is passed, each type will be assumed to be Any.
+        metrics (List[str]): List of names of metrics to be used in the task.
+        prediction_type (Optional[Any]):
+            Need to be consistent with all used metrics. Defaults to None, which means that it will
+            be inferred based on prediction type of specified metrics.
+
     The output instance contains three fields:
         "inputs" whose value is a sub-dictionary of the input instance, consisting of all the fields listed in Arg 'inputs'.
         "outputs" -- for the fields listed in Arg "outputs".
         "metrics" -- to contain the value of Arg 'metrics'
-
     """
 
-    inputs: List[str]
-    outputs: List[str]
+    inputs: Union[Dict[str, Any], List[str]]
+    outputs: Union[Dict[str, Any], List[str]]
     metrics: List[str]
+    prediction_type: Optional[Any] = None
     augmentable_inputs: List[str] = []
 
+    @staticmethod
+    def process_type(data_type: Any) -> Any:
+        if isinstance(data_type, str):
+            data_type = data_type.replace("typing.", "")
+            return parse_type_string(data_type)
+        return data_type
+
     def verify(self):
+        for io_type in ["inputs", "outputs"]:
+            data = self.inputs if io_type == "inputs" else self.outputs
+            if not isinstance(data, Dict):
```

Comment:
```suggestion
            if not isinstance(data, Dict[str,str]):
```

Reply:
I intended to make it more general so that a user can pass a direct type as well, not only type strings. For example: 
`{"some_field": List[str]}` would be also valid instead of only `{"some_field": "List[str]"}`. In that case, the `data` from above would be `Dict[str,Any]`.
What do you think? Or should I change it so that types are specified as strings only?

Reply:
I also opted to use the actual type,  but it does not work (I had the same issue with the metrics).  This is because Dict[str,Any] is not serializable and hence would not be written to the json correctly.   So. I suggest using only string.